File: front/src/app/features/articles/interfaces/comment.ts
/**
 * Interface representing a comment on an article.
 * 
 * The `Comment` interface defines the structure of a comment object, which includes details
 * about the comment itself, the author, and the article to which it belongs. It captures
 * information such as the comment's unique identifier, its content, the author's name,
 * and timestamps for creation and last update.
 * 
 * @interface Comment
 * @property {number} id - The unique identifier of the comment. This field is required.
 * @property {string} content - The text content of the comment. This field is required.
 * @property {string} author - The name of the author who created the comment. This field is required.
 * @property {string} article_id - The unique identifier of the article that the comment is associated with. This field is required.
 * @property {string} [createdAt] - The date and time when the comment was created. This field is optional.
 * @property {string} [updatedAt] - The date and time when the comment was last updated. This field is optional.
 */
export interface Comment {
    id: number;
    content: string;
    author: string;
    article_id: string;
    createdAt?: string;
    updatedAt?: string;
}

File: back/src/main/java/com/openclassrooms/mddapi/models/Comment.java
package com.openclassrooms.mddapi.models;

import java.time.LocalDateTime;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import com.fasterxml.jackson.annotation.JsonManagedReference;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EntityListeners;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.experimental.Accessors;

/**
 * Represents a comment associated with an article.
 * A comment contains the content of the comment, the article it relates to,
 * the user who authored the comment, and timestamps for creation and last update.
 */
@Entity
@EntityListeners(AuditingEntityListener.class)
@Table(name = "COMMENTS")
@Data
@Accessors(chain = true)
@EqualsAndHashCode(of = {"id"})
@Builder
@NoArgsConstructor
@AllArgsConstructor
@ToString
public class Comment {

    /**
     * Unique identifier for the comment.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    /**
     * Content of the comment.
     * Must not be null and cannot exceed 2500 characters.
     */
    @NotNull
    @Size(max = 2500)
    private String content;
    
    /**
     * The article to which this comment belongs.
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "article_id", nullable = false)
    @JsonManagedReference
    private Article article;
    
    /**
     * The user who authored the comment.
     */
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author", nullable = false)
    @JsonManagedReference
    private User author;
    
    /**
     * Date and time when the comment was created.
     * This field is automatically set by the system and is not updatable.
     */
    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    /**
     * Date and time when the comment was last updated.
     */
    @LastModifiedDate
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
